org.alloytools.kodkod.core/src/main/java/kodkod/engine/solveengine/HttpClient.java

```diff
@@ -0,0 +1,312 @@
+package kodkod.engine.solveengine;
+
+import kodkod.engine.solveengine.enums.SEStatusCode;
+
+import javax.json.bind.Jsonb;
+import javax.json.bind.JsonbBuilder;
+import javax.json.bind.spi.JsonbProvider;
+import javax.net.ssl.HttpsURLConnection;
+import java.io.*;
+import java.lang.reflect.Type;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.net.UnknownHostException;
+import java.rmi.ServerException;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.concurrent.TimeUnit;
+
+public class HttpClient {
+    /** Containing the problem to solve. */
+    private Model model;
+    /** Amount of seconds to wait between each fetchStatus request */
+    private int sleepTime;
+    /** The id of the job that is created when the problem is sent to SolveEngine
+     * Allows to retrieve the problem being solved */
+    private String jobId;
+    /** Booleans to figure out if this instance has already been launched but stopped before the end*/
+    private boolean jobCreated = false;
+    private boolean jobScheduled = false;
+    private boolean jobDone = false;
+    /** Updated status of the job processing the problem*/
+    private SEStatusCode seStatus = SEStatusCode.NOTSTARTED;
+
+    public int getSleepTime() { return sleepTime; }
+    public void setSleepTime(int sleepTime) { this.sleepTime = sleepTime; }
```

Comment:
this part is taken from the actual Java SolveEngine module that is getting prepared
should I remove it or leave it in case it could be useful later ? (in the future, all this code could be directly imported from a an external module, instead of weighing your software)